File: Examples/Objective-C/ConnectHelp-ObjC/ViewController.m
//
//  ViewController.m
//  ConnectHelp-ObjC
//
//  Created by Matthew Lewin on 3/14/16.
//  Copyright © 2016 Boomtown. All rights reserved.
//

#import "ViewController.h"

NSString *apiToken   	= @"__API_TOKEN__";
NSString *apiSecret   	= @"__API_SECRET_KEY__";

NSString *memberID          = @"__MEMBERS_ID__";
NSString *memberUserID     	= @"__MEMBERS_USERS_ID__";
NSString *memberLocationID 	= @"__MEMBERS_LOCATIONS_ID__";

@interface ViewController () <BTConnectHelpButtonDelegate>

@property (weak, nonatomic) IBOutlet BTConnectHelpButton    *storyboardHelpButton;
@property (weak, nonatomic) IBOutlet UIView                 *subview;

@property (strong, nonatomic)	BTConnectHelpButton *programmaticHelpButton;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = @"Boomtown Sample App";
    
    [self.storyboardHelpButton setCredentialsWithToken:apiToken secret:apiSecret];

    self.storyboardHelpButton.delegate = self;

    self.storyboardHelpButton.memberID         	= memberID;
	self.storyboardHelpButton.memberUserID      = memberUserID;
    self.storyboardHelpButton.memberLocationID  = memberLocationID;
	
    self.storyboardHelpButton.supportWebsiteURL   = [NSURL URLWithString:@"http://example.com"];
    self.storyboardHelpButton.supportEmailAddress = @"support@example.com";
    self.storyboardHelpButton.supportPhoneNumber  = @"1-888-555-2368";
	
	self.programmaticHelpButton = [[BTConnectHelpButton alloc] initWithFrame:self.view.frame];
	self.programmaticHelpButton.delegate = self;
	
	self.programmaticHelpButton.memberID           	= memberID;
	self.programmaticHelpButton.memberUserID      	= memberUserID;
	self.programmaticHelpButton.memberLocationID 	= memberLocationID;
	
	self.programmaticHelpButton.supportWebsiteURL 	= [NSURL URLWithString:@"http://example.com"];
	self.programmaticHelpButton.supportEmailAddress = @"support@example.com";
	self.programmaticHelpButton.supportPhoneNumber 	= @"1-888-555-2368";
	[self.programmaticHelpButton setCredentialsWithToken:apiToken secret:apiSecret];
	[self.view addSubview:self.programmaticHelpButton];
    
    [self.programmaticHelpButton advertiseServiceWithPublicData:@{@"test":@"data"}
                                                    privateData:@{@"test":@"private data"}];

}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
}


#pragma mark - Help button delegates

- (void) helpButton:(BTConnectHelpButton *)helpButton didAdvertiseService:(NSNetService *)netService
{
    NSLog(@"mDNS service successfully advertised.");
}


- (void) helpButton:(BTConnectHelpButton *)helpButton didFailToAdvertiseService:(NSDictionary<NSString *,NSNumber *> *)errorDict
{
    NSLog(@"mDNS service failed to advertise.");
}




- (void) helpButton:(nonnull BTConnectHelpButton *)helpButton  displayHelpActionSheet:(nonnull UIAlertController *)alertController
{
    if ( self.view.traitCollection.horizontalSizeClass != UIUserInterfaceSizeClassCompact )
    {
        [self presentViewController:alertController animated:YES completion:nil];
		UIPopoverPresentationController *popPresenter = [alertController
														 popoverPresentationController];
		popPresenter.sourceView = helpButton;
		popPresenter.sourceRect = helpButton.bounds;
	}
	else
	{
		[self.navigationController presentViewController:alertController animated:YES completion:nil];
	}
}


- (void) helpButton:(nonnull BTConnectHelpButton *)helpButton displayIssueViewController:(UIViewController *)viewController
{
	[self.navigationController pushViewController:viewController animated:YES];
}


- (void) helpButton:(nonnull BTConnectHelpButton *)helpButton didFailWithError:(nonnull NSError *)error
{
	NSLog(@"%@: %@", error.localizedDescription, error.localizedFailureReason);
}


- (void) helpButtonDidSetCredentials:(nonnull BTConnectHelpButton *)helpButton
{
    NSLog(@"Everything is ready to proceed.");
}




@end

File: Examples/Swift/ConnectHelp-Swift/ViewController.swift
//
//  ViewController.swift
//  ConnectHelp-Swift
//
//  Created by Matthew Lewin on 3/18/16.
//  Copyright © 2016 Boomtown. All rights reserved.
//

import UIKit
import BTConnectHelp

let apiToken            = "__API_TOKEN__"
let apiSecret           = "__API_SECRET_KEY__"

let memberID            = "__MEMBERS_ID__"
let memberUserID        = "__MEMBERS_USERS_ID__"
let memberLocationID    = "__MEMBERS_LOCATIONS_ID__"

class ViewController: UIViewController, BTConnectHelpButtonDelegate {

    @IBOutlet weak var subview: UIView!
    @IBOutlet weak var storyboardHelpButton: BTConnectHelpButton! {
        didSet {
            storyboardHelpButton.setCredentialsWithToken(apiToken, secret: apiSecret)
			
            storyboardHelpButton.delegate = self
            
            storyboardHelpButton.memberID          	= memberID
			storyboardHelpButton.memberUserID      	= memberUserID
            storyboardHelpButton.memberLocationID  	= memberLocationID
			
            storyboardHelpButton.supportWebsiteURL 	 = NSURL(string: "http://example.com") as URL?;
            storyboardHelpButton.supportEmailAddress = "support@example.com"
            storyboardHelpButton.supportPhoneNumber  = "1-888-555-2368"
        }
    }
    
    lazy var programmaticHelpButton: BTConnectHelpButton = {
        [unowned self] in
        let button = BTConnectHelpButton(frame: self.view.frame)
        button.delegate = self
        
        button.memberID          = memberID
        button.memberUserID      = memberUserID
        button.memberLocationID  = memberLocationID

        button.supportWebsiteURL   =  URL(string: "http://example.com")
        button.supportEmailAddress = "support@example.com"
        button.supportPhoneNumber  = "1-888-555-2368"
        
        button.setCredentialsWithToken(apiToken, secret: apiSecret)
		
        return button
    }()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = "Boomtown Sample App"
        self.view.addSubview(programmaticHelpButton)
    }

    override func viewDidAppear(_ animated: Bool) {
        super.viewDidAppear(animated)
    }

    func helpButton(_ helpButton: BTConnectHelpButton, displayIssueViewController viewController: UIViewController) {
        show(viewController, sender: helpButton)
    }

    func helpButton(_ helpButton: BTConnectHelpButton, displayHelpActionSheet alertController: UIAlertController) {
        present(alertController, animated: true, completion: nil)
        if let popoverController = alertController.popoverPresentationController {
            popoverController.sourceView = helpButton
            popoverController.sourceRect = helpButton.bounds
        }
	}
	
    func helpButton(_ helpButton: BTConnectHelpButton, didFailWithError error: Error) {
        NSLog("%@", error.localizedDescription)
    }
	
    func helpButtonDidSetCredentials(_ helpButton: BTConnectHelpButton) {
        NSLog("Everything is ready to proceed.")
    }
    
}
